feat(navbar): automatically close nav bar when item is clicked

EnigmaCurry opened the pull request:
<!--
There are two main goals in this document, depending on the nature of your PR:

- description: please tell us about your PR
- checklist: please review the checklist

To help to quickly understand the nature of your pull request,
please create a description that incorporates the following elements:
-->

### What is accomplished by your PR?
Closes the nav bar automatically once an item in is clicked.

### Is there something controversial in your PR?
I'm using a global event bus, because the VueNavBar and the items in it are in disparate components and they need a way to talk to eachother. This doesn't feel like the best way to me, but I'm a VueJS beginner, so I would love to learn the better way!

# Checklist

### New Feature / Bug Fix

- [x] Run unit tests to ensure all existing tests are still passing
- [ ] Add new passing unit tests to cover the code introduced by your PR

I have added a test, but it does not work. I can't find the element to click on to test it. Perhaps this cannot be unit tested, because it is implemented across components? Any insight into how to approach this is appreciated!

<!--
Thanks for contributing!
-->

devCrossNet commented on src/app/event-bus.ts:
```diff
@@ -0,0 +1,5 @@
+import Vue                from 'vue';
```
we already have an event bus: https://github.com/devCrossNet/vue-starter/blob/master/src/app/shared/services/EventBus.ts

devCrossNet commented on src/app/App.vue:
```diff
@@ -75,7 +76,11 @@
+        this.navBarClose();
       },
+      navBarClose() {
+          EventBus.$emit('closeNavBar');
```
I would prefer a namespaced event naming like https://github.com/devCrossNet/vue-starter/blob/master/src/app/shared/components/VueNotificationStack/VueNotificationStack.vue#L60

devCrossNet commented on src/app/shared/components/VueNavBar/VueNavBar.spec.ts:
```diff
@@ -80,4 +80,27 @@ describe('VueNavBar.vue', () => {
   });
+
+  test('should close menu when item is clicked', () => {
+    const wrapper: any = mount(VueNavBar, {
```
you could write the test like this:

- initialize the wrapper with isOpen
- use the global event bus to trigger the close event
- assert if wrapper.vm.isOpen is false

this would be independent of the child component.

if you want to do it more like an integration test, you have to provide a slot with some markup or a test component which triggers the event bus and then find the right html element to trigger the click event. 

the navbar itself has no links in it (as you see in App.vue), it just provides a slot for a child component.

EnigmaCurry commented:
@devCrossNet Thank you for the helpful feedback!

Changes since last time:
 * Refactored code based on your suggestions.
 * Reimplemented test by interfacing with the EventBus.
 * Fresh rebase off of your master branch.
 * Force pushed to my branch.
